File: decompiled/sources/com/hjq/permissions/PermissionApi.java
package com.hjq.permissions;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import java.util.ArrayList;
import java.util.List;

final class PermissionApi {
    private static final PermissionDelegate DELEGATE;

    PermissionApi() {
    }

    static {
        if (AndroidVersion.isAndroid13()) {
            DELEGATE = new PermissionDelegateImplV33();
        } else if (AndroidVersion.isAndroid12()) {
            DELEGATE = new PermissionDelegateImplV31();
        } else if (AndroidVersion.isAndroid11()) {
            DELEGATE = new PermissionDelegateImplV30();
        } else if (AndroidVersion.isAndroid10()) {
            DELEGATE = new PermissionDelegateImplV29();
        } else if (AndroidVersion.isAndroid9()) {
            DELEGATE = new PermissionDelegateImplV28();
        } else if (AndroidVersion.isAndroid8()) {
            DELEGATE = new PermissionDelegateImplV26();
        } else if (AndroidVersion.isAndroid6()) {
            DELEGATE = new PermissionDelegateImplV23();
        } else {
            DELEGATE = new PermissionDelegateImplV14();
        }
    }

    static boolean isGrantedPermission(Context context, String str) {
        return DELEGATE.isGrantedPermission(context, str);
    }

    static boolean isPermissionPermanentDenied(Activity activity, String str) {
        return DELEGATE.isPermissionPermanentDenied(activity, str);
    }

    static Intent getPermissionIntent(Context context, String str) {
        return DELEGATE.getPermissionIntent(context, str);
    }

    static boolean isSpecialPermission(String str) {
        return PermissionUtils.isSpecialPermission(str);
    }

    static boolean containsSpecialPermission(List<String> list) {
        if (list != null && !list.isEmpty()) {
            for (String isSpecialPermission : list) {
                if (isSpecialPermission(isSpecialPermission)) {
                    return true;
                }
            }
        }
        return false;
    }

    static boolean isGrantedPermissions(Context context, List<String> list) {
        if (list == null || list.isEmpty()) {
            return false;
        }
        for (String isGrantedPermission : list) {
            if (!isGrantedPermission(context, isGrantedPermission)) {
                return false;
            }
        }
        return true;
    }

    static List<String> getGrantedPermissions(Context context, List<String> list) {
        ArrayList arrayList = new ArrayList(list.size());
        for (String next : list) {
            if (isGrantedPermission(context, next)) {
                arrayList.add(next);
            }
        }
        return arrayList;
    }

    static List<String> getDeniedPermissions(Context context, List<String> list) {
        ArrayList arrayList = new ArrayList(list.size());
        for (String next : list) {
            if (!isGrantedPermission(context, next)) {
                arrayList.add(next);
            }
        }
        return arrayList;
    }

    static boolean isPermissionPermanentDenied(Activity activity, List<String> list) {
        for (String isPermissionPermanentDenied : list) {
            if (isPermissionPermanentDenied(activity, isPermissionPermanentDenied)) {
                return true;
            }
        }
        return false;
    }

    static List<String> getDeniedPermissions(List<String> list, int[] iArr) {
        ArrayList arrayList = new ArrayList();
        for (int i = 0; i < iArr.length; i++) {
            if (iArr[i] == -1) {
                arrayList.add(list.get(i));
            }
        }
        return arrayList;
    }

    static List<String> getGrantedPermissions(List<String> list, int[] iArr) {
        ArrayList arrayList = new ArrayList();
        for (int i = 0; i < iArr.length; i++) {
            if (iArr[i] == 0) {
                arrayList.add(list.get(i));
            }
        }
        return arrayList;
    }
}

File: decompiled/sources/com/hjq/permissions/PermissionApi.kt
package com.hjq.permissions

import android.app.Activity
import android.content.Context
import android.content.Intent
import java.util.ArrayList

object PermissionApi {
    private val DELEGATE: PermissionDelegate =
        when {
            AndroidVersion.isAndroid13() -> PermissionDelegateImplV33()
            AndroidVersion.isAndroid12() -> PermissionDelegateImplV31()
            AndroidVersion.isAndroid11() -> PermissionDelegateImplV30()
            AndroidVersion.isAndroid10() -> PermissionDelegateImplV29()
            AndroidVersion.isAndroid9() -> PermissionDelegateImplV28()
            AndroidVersion.isAndroid8() -> PermissionDelegateImplV26()
            AndroidVersion.isAndroid6() -> PermissionDelegateImplV23()
            else -> PermissionDelegateImplV14()
        }

    @JvmStatic
    fun isGrantedPermission(context: Context, permission: String): Boolean =
        DELEGATE.isGrantedPermission(context, permission)

    @JvmStatic
    fun isPermissionPermanentDenied(activity: Activity, permission: String): Boolean =
        DELEGATE.isPermissionPermanentDenied(activity, permission)

    @JvmStatic
    fun getPermissionIntent(context: Context, permission: String): Intent =
        DELEGATE.getPermissionIntent(context, permission)

    @JvmStatic
    fun isSpecialPermission(permission: String): Boolean =
        PermissionUtils.isSpecialPermission(permission)

    @JvmStatic
    fun containsSpecialPermission(permissions: List<String>?): Boolean {
        if (permissions.isNullOrEmpty()) {
            return false
        }
        for (permission in permissions) {
            if (isSpecialPermission(permission)) {
                return true
            }
        }
        return false
    }

    @JvmStatic
    fun isGrantedPermissions(context: Context, permissions: List<String>?): Boolean {
        if (permissions.isNullOrEmpty()) {
            return false
        }
        for (permission in permissions) {
            if (!isGrantedPermission(context, permission)) {
                return false
            }
        }
        return true
    }

    @JvmStatic
    fun getGrantedPermissions(context: Context, permissions: List<String>): List<String> {
        val grantedPermissions = ArrayList<String>(permissions.size)
        for (permission in permissions) {
            if (isGrantedPermission(context, permission)) {
                grantedPermissions.add(permission)
            }
        }
        return grantedPermissions
    }

    @JvmStatic
    fun getDeniedPermissions(context: Context, permissions: List<String>): List<String> {
        val deniedPermissions = ArrayList<String>(permissions.size)
        for (permission in permissions) {
            if (!isGrantedPermission(context, permission)) {
                deniedPermissions.add(permission)
            }
        }
        return deniedPermissions
    }

    @JvmStatic
    fun isPermissionPermanentDenied(activity: Activity, permissions: List<String>): Boolean {
        for (permission in permissions) {
            if (isPermissionPermanentDenied(activity, permission)) {
                return true
            }
        }
        return false
    }

    @JvmStatic
    fun getDeniedPermissions(permissions: List<String>, grantResults: IntArray): List<String> {
        val deniedPermissions = ArrayList<String>()
        for (index in grantResults.indices) {
            if (grantResults[index] == -1) {
                deniedPermissions.add(permissions[index])
            }
        }
        return deniedPermissions
    }

    @JvmStatic
    fun getGrantedPermissions(permissions: List<String>, grantResults: IntArray): List<String> {
        val grantedPermissions = ArrayList<String>()
        for (index in grantResults.indices) {
            if (grantResults[index] == 0) {
                grantedPermissions.add(permissions[index])
            }
        }
        return grantedPermissions
    }
}
